iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -747,31 +747,54 @@ private async Task EnsureOpenedAsync(TimeoutHelper timeoutHelper)
             }
         }
 
-        private Task OpenInternalAsync(CancellationToken cancellationToken)
+        private async Task OpenInternalAsync(bool withRetry, CancellationToken cancellationToken)
```

Comment:
Previously this method returned a Task, but that task was immediately awaited wherever it was called, so making this async should be safe